tests/foreman/installer/test_installer.py

```diff
@@ -212,25 +214,155 @@ def test_capsule_installation(sat_non_default_install, cap_ready_rhel, setting_u
 
     :customerscenario: true
     """
-    # Get Capsule repofile, and enable and download satellite-capsule
+    # Create testing organization
     org = sat_non_default_install.api.Organization().create()
-    cap_ready_rhel.register_to_cdn()
-    cap_ready_rhel.download_repofile(
-        product='capsule',
-        release=settings.server.version.release,
-        snap=settings.server.version.snap,
+
+    # Unregister capsule in case it's registered to CDN
+    cap_ready_rhel.unregister()
+
+    # Add a manifest to the Satellite
+    with Manifester(manifest_category=settings.manifest.entitlement) as manifest:
+        sat_non_default_install.upload_manifest(org.id, manifest.content)
+    # Create capsule certs and activation key
+    file, _, cmd_args = sat_non_default_install.capsule_certs_generate(cap_ready_rhel)
+    sat_non_default_install.session.remote_copy(file, cap_ready_rhel)
+    ak = sat_non_default_install.api.ActivationKey(
+        organization=org, environment=org.library
+    ).create()
+
+    # List of sync tasks - all repos will be synced asynchronously
+    sync_tasks = []
+
+    # Enable and sync RHEL BaseOS and AppStream repos
+    if settings.robottelo.rhel_source == "internal":
+        # Configure internal sources as custom repositories
+        product_rhel = sat_non_default_install.api.Product(organization=org.id).create()
+        for repourl in settings.repos.get(f'rhel{cap_ready_rhel.os_version.major}_os').values():
+            repo = sat_non_default_install.api.Repository(
+                organization=org.id, product=product_rhel, content_type='yum', url=repourl
+            ).create()
+            # custom repos need to be explicitly enabled
+            ak.content_override(
+                data={
+                    'content_overrides': [
+                        {
+                            'content_label': '_'.join([org.label, product_rhel.label, repo.label]),
+                            'value': '1',
+                        }
+                    ]
+                }
+            )
```

Comment:
I moved the steps to prepare the repositories into separate function, so it can be re-used in other places. But I think it's not necessary to move it outside of this test suite, because this is the only place where we observe the installation process. (Other tests use templates with pre-installed Satellite/Capsule) .
Luckily the content override works "one by one", so all the repositories are enabled at the end. Performance impact is negligible - the API call takes just few milliseconds.

Reply:
I don't know the robottelo code base well (but am familiar with pytest). Is this something you'd use as a fixture? The reuse is probably easier, but I can also see some downsides in debugging (heavy fixtures that fail are often a pain to debug).